docs/database/schema.sql

```diff
@@ -103,3 +103,28 @@ CREATE TABLE action_instance
     CONSTRAINT action_instance_configured_connector_id_fk FOREIGN KEY (configured_connector_id) REFERENCES configured_connector (id)
 );
 
+-- An integration is the integration definition. It contains a list of integration steps.
+CREATE TABLE integration
+(
+    id SERIAL PRIMARY KEY NOT NULL,
+    name VARCHAR(256) NOT NULL,
+    description VARCHAR(2048)
+);
+CREATE UNIQUE INDEX integration_name_uindex ON integration (name);
+
+-- An integration step is the definition of a single step in an integration. It contains configuration
+-- for each and every step in the integration flow.
+-- This can reference an action instance or be a supported integration pattern (e.g. log, wiretap, choice, etc).
+CREATE TABLE integration_step
+(
+    id SERIAL PRIMARY KEY NOT NULL,
+    integration_id INTEGER NOT NULL,
+    action_instance_id INTEGER,
```

Comment:
So I guess for intermediate steps this can be `null` ? Can an intermediate step refer an action  ? Probably not, so maybe we can model this in Java as `ActionStep extends Step` and leave the action_instance_id null for the base class.